File: content-solutions/queue-deque/queue-round-kitchen.js
function solution(dishes, q) {
  const line = dishes.map(([name, time]) => [name, time]);
  let head = 0;
  const done = [];
  while (head < line.length) {
    const [name, left] = line[head++];
    if (left - q <= 0) done.push(name);
    else line.push([name, left - q]);
  }
  return done;
}

File: content-solutions/queue-deque/queue-round-kitchen.py
from collections import deque


def solution(dishes, q):
    line = deque((name, time) for name, time in dishes)
    done = []
    while line:
        name, left = line.popleft()
        left -= q
        if left <= 0:
            done.append(name)
        else:
            line.append((name, left))
    return done
